pkg/cache/internal/cache_reader.go

```diff
@@ -73,14 +73,20 @@ func (c *CacheReader) Get(_ context.Context, key client.ObjectKey, out runtime.O
 	// TODO(directxman12): revisit the decision to always deepcopy
 	obj = obj.(runtime.Object).DeepCopyObject()
 
-	// Copy the value of the item in the cache to the returned value
-	// TODO(directxman12): this is a terrible hack, pls fix (we should have deepcopyinto)
-	outVal := reflect.ValueOf(out)
-	objVal := reflect.ValueOf(obj)
-	if !objVal.Type().AssignableTo(outVal.Type()) {
-		return fmt.Errorf("cache had type %s, but %s was asked for", objVal.Type(), outVal.Type())
+	oType := reflect.TypeOf(obj)
+	_, ok := oType.MethodByName("DeepCopyInto")
+	if ok {
```

Comment:
Would you pull this logic into its own function and unit test it?  Verify both code paths work?

Reply:
The first one.